core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2665,8 +2665,25 @@ object SparkContext extends Logging {
     // When running locally, don't try to re-execute tasks on failure.
     val MAX_LOCAL_TASK_FAILURES = 1
 
+    // SPARK-26340: Ensure that executor's core num meets at least one task requirement.
+    def checkCpusPerTask(
+      executorCoreNum: Int = sc.conf.get(EXECUTOR_CORES),
+      clusterMode: Boolean = true): Unit = {
+      val cpusPerTask = sc.conf.get(CPUS_PER_TASK)
+      if (executorCoreNum < cpusPerTask) {
+        val message = if (clusterMode) {
```

Comment:
I don't think you need two messages here. Just have one stating that `${CPUS_PER_TASK.key} must be <= the $executorCoreNum cores available per executor`

Reply:
`executorCoreNum` is decided by `local[N]` or `--conf spark.executor.cores=M`, I think we should info this in exception.
And, if user both sets `--master local[N]` and `--conf spark.executor.cores=M`, I think we should ignore the latter.